src/app/shared/element-child-data-classes-child-elements-list/flattened-data-classes-list.component.ts

```diff
@@ -0,0 +1,241 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import {
+  Component,
+  Input,
+  ViewChildren,
+  ViewChild,
+  AfterViewInit,
+  ElementRef,
+  EventEmitter,
+  ChangeDetectorRef,
+  OnInit,
+  Output
+} from '@angular/core';
+import { StateHandlerService } from '@mdm/services/handlers/state-handler.service';
+import { MdmResourcesService } from '@mdm/modules/resources';
+import { merge, Observable } from 'rxjs';
+import { catchError, map, startWith, switchMap } from 'rxjs/operators';
+import { MatSort } from '@angular/material/sort';
+import { MdmPaginatorComponent } from '../mdm-paginator/mdm-paginator';
+import { MatDialog } from '@angular/material/dialog';
+import { BulkDeleteModalComponent } from '@mdm/modals/bulk-delete-modal/bulk-delete-modal.component';
+import { GridService } from '@mdm/services/grid.service';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { MatTable } from '@angular/material/table';
+import { MessageHandlerService } from '@mdm/services';
+import { DataClass } from '@maurodatamapper/mdm-resources';
+
+@Component({
+  selector: 'mdm-flattened-data-classes-list',
+  templateUrl: './flattened-data-classes-list.component.html',
+  styleUrls: ['./flattened-data-classes-list.component.sass']
+})
+export class FlattenedDataClassesComponent implements AfterViewInit, OnInit {
+  @Input() parentDataModel: any;
+  @Input() parentDataClass: any;
+  @Input() mcDataClass: any;
+  @Input() type: any;
+  @Input() childDataClasses: any;
+  @Input() isEditable: any;
+  @Output() totalCount = new EventEmitter<string>();
+
+  @ViewChildren('filters', { read: ElementRef }) filters: ElementRef[];
+  @ViewChild(MatSort, { static: false }) sort: MatSort;
+  @ViewChild(MdmPaginatorComponent, { static: true }) paginator: MdmPaginatorComponent;
+  @ViewChild(MatTable, { static: false }) table: MatTable<any>;
+
+  processing: boolean;
+  failCount: number;
+  total: number;
+  showStaticRecords: any;
+  dataSource: any;
+  records: any[] = [];
+  hideFilters = true;
+  displayedColumns: string[];
+  loading: boolean;
+  totalItemCount = 0;
+  isLoadingResults = true;
+  filterEvent = new EventEmitter<any>();
+  filter: {};
+  deleteInProgress: boolean;
+  checkAllCheckbox = false;
+  bulkActionsVisible = 0;
+
+  constructor(
+    private changeRef: ChangeDetectorRef,
+    private resources: MdmResourcesService,
+    private stateHandler: StateHandlerService,
+    private dialog: MatDialog,
+    private gridService: GridService,
+    private messageHandler: MessageHandlerService,
+  ) { }
+
+  ngOnInit(): void {
+    if (this.isEditable && !this.parentDataModel.finalised) {
```

Comment:
Whether the model is finalised or not, the same columns will be need to be visible - Data Element, Data Class, Description.